### PR TITLE
Implements #4360

### PR DESCRIPTION
**Pull request by kerberjg**
Makes [BoundingBox](https://libgdx.badlogicgames.com/nightlies/docs/api/com/badlogic/gdx/math/collision/BoundingBox.html) compatible with 2D by performing a "*greater or equal*" comparison between `min` and `max` in BoundingBox#isValid().

Naturally, this would cause a BoundingBox with an area of 0 to be considered valid, so a `&& !dim.isZero()` was added to the expression as well in order to prevent that.

EDIT: updated the dim size check

### CODE REVIEW

**Comment by Darkyenus** [comment]
Am I missing something or would this mean that only zero-size bounding box is considered valid? Shouldn't it be `!dim.isZero()`?

**Comment by kerberjg** [comment]
@Darkyenus You're right, it was a typo. Just fixed it.

**Comment by xoppa** [comment]
Why did you make an exception for at least one axis to be not zero? If a zero volume (3D) is valid and a zero area (2D) is a valid then I guess that a zero length (1D) should be valid as well.

**Comment by kerberjg** [comment]
@xoppa I actually had in mind that none of them are to be considered valid, but I guess that's a mistake. Upon further consideration it could be considered valid indeed.

File: gdx/src/com/badlogic/gdx/math/collision/BoundingBox.java [hunk 1]
```diff
@@ -210,7 +210,7 @@ public BoundingBox clr () {
 	/** Returns whether this bounding box is valid. This means that {@link #max} is greater than or equal to {@link #min}, with {@link #dim} greater than zero.
```
Comment (moonwolf-github) [inline gdx/src/com/badlogic/gdx/math/collision/BoundingBox.java]: I think you should update comment also.

Comment (kerberjg) [inline gdx/src/com/badlogic/gdx/math/collision/BoundingBox.java]: Done now

**Comment by xoppa** [comment]
Thanks

**Pull request merged**